### PR TITLE
Update 0.1.1

### PR DESCRIPTION
**Pull request by Shawdooow**

### CODE REVIEW

File: osu.Game.Rulesets.Vitaru/Objects/Drawables/DrawableCharacter.cs [hunk 1]
```diff
@@ -40,7 +40,7 @@ public abstract class DrawableCharacter : DrawableVitaruHitObject
         protected float HitboxWidth { get; set; }
 
         public Action OnDeath { get; set; }
-        public Action OnShoot { get; set; }
+        public Action CharacterShoot { get; set; }
```
Comment (Jorolf) [inline osu.Game.Rulesets.Vitaru/Objects/Drawables/DrawableCharacter.cs]: why did you change the name?

File: osu.Game.Rulesets.Vitaru/Objects/Drawables/DrawableVitaruEnemy.cs [hunk 2]
```diff
@@ -120,6 +116,14 @@ protected override void UpdateState(ArmedState state)
             Expire();
         }
 
+        private void leave()
+        {
+            int r = RNG.Next(-100, 612);
+            MoveTo(new Vector2(r, -300), 2000, EasingTypes.InCubic);
```
Comment (Jorolf) [inline osu.Game.Rulesets.Vitaru/Objects/Drawables/DrawableVitaruEnemy.cs]: why do you have to move it when it leaves?

File: osu.Game.Rulesets.Vitaru/Objects/Drawables/DrawableCharacter.cs (hunk 1 above)
Comment (Shawdooow) [inline osu.Game.Rulesets.Vitaru/Objects/Drawables/DrawableCharacter.cs]: OnShoot was a weird name, CharacterShoot makes more sense imo

File: osu.Game.Rulesets.Vitaru/Objects/Drawables/DrawableVitaruEnemy.cs (hunk 2 above)
Comment (Shawdooow) [inline osu.Game.Rulesets.Vitaru/Objects/Drawables/DrawableVitaruEnemy.cs]: Because it is better than phasing out of existence?

File: osu.Game.Rulesets.Vitaru/Objects/Drawables/DrawableVitaruPlayer.cs [hunk 3]
```diff
@@ -61,57 +66,59 @@ protected override void Update()
...
+
+        private void playerInput()
+        {
+            if (keys[Key.Z])
             {
-                pos.X -= xSpeed;
+                Shooting = true;
             }
-            if (keys[Key.Down])
+            if (keys[Key.Z] == false)
```
Comment (Jorolf) [inline osu.Game.Rulesets.Vitaru/Objects/Drawables/DrawableVitaruPlayer.cs]: Shooting = keys[Key.Z]

Comment (Shawdooow) [inline osu.Game.Rulesets.Vitaru/Objects/Drawables/DrawableVitaruPlayer.cs]: ?

Comment (Jorolf) [inline osu.Game.Rulesets.Vitaru/Objects/Drawables/DrawableVitaruPlayer.cs]: rather than doing a if statement you could just do `Shooting = keys[Key.Z]`

Comment (Jorolf) [inline osu.Game.Rulesets.Vitaru/Objects/Drawables/DrawableVitaruPlayer.cs]: or just move that to `OnKeyPress`

Comment (Shawdooow) [inline osu.Game.Rulesets.Vitaru/Objects/Drawables/DrawableVitaruPlayer.cs]: That wouldn't allow me to change the pattern being thrown as easily. It would also look messier 🤔

Comment (Jorolf) [inline osu.Game.Rulesets.Vitaru/Objects/Drawables/DrawableVitaruPlayer.cs]: why are you changing the pattern with keys[Key.Z]???

Comment (Shawdooow) [inline osu.Game.Rulesets.Vitaru/Objects/Drawables/DrawableVitaruPlayer.cs]: I'm not?

Comment (Jorolf) [inline osu.Game.Rulesets.Vitaru/Objects/Drawables/DrawableVitaruPlayer.cs]: then what is the problem with `Shooting = keys[Key.Z]`

Comment (Shawdooow) [inline osu.Game.Rulesets.Vitaru/Objects/Drawables/DrawableVitaruPlayer.cs]: Because the player wont always throw the exact same pattern. What if I want it to throw the spinner one for whatever reason? Or let mappers pick their own player patterns?

Comment (Jorolf) [inline osu.Game.Rulesets.Vitaru/Objects/Drawables/DrawableVitaruPlayer.cs]: then change that in private void shoot and not there smh

Comment (Shawdooow) [inline osu.Game.Rulesets.Vitaru/Objects/Drawables/DrawableVitaruPlayer.cs]: Hmm, okay

**Comment by Shawdooow** [comment]
Also gotta finish patterns, totally forgot half of them

File: osu.Game.Rulesets.Vitaru/Objects/Drawables/DrawableVitaruPlayer.cs (hunk 3 above)
Comment (Shawdooow) [inline osu.Game.Rulesets.Vitaru/Objects/Drawables/DrawableVitaruPlayer.cs]: Gonna put this whole situation on hold and just do a `KeyConversionManager` next Update

File: osu.Game.Rulesets.Vitaru/Objects/Drawables/DrawableVitaruEnemy.cs [hunk 4]
```diff
@@ -197,94 +197,87 @@ private void spinner()
...
-                            directionModifier + randomDirection
-                        );
-                        speedModifier -= 0.01f;
-                        directionModifier += 0.075f;
-                    }
+                        Origin = Anchor.Centre,
+                        Depth = 6,
+                        PatternColor = Color4.Green,
+                        PatternAngleRadian = playerPos,
+                        PatternSpeed = 0.4f,
```
Comment (Arrcival) [inline osu.Game.Rulesets.Vitaru/Objects/Drawables/DrawableVitaruEnemy.cs]: Change PatternSpeed to 0.1f

File: osu.Game.Rulesets.Vitaru/Objects/Projectiles/BulletPatterns.cs [hunk 5]
```diff
@@ -7,61 +7,185 @@
...
+        public CoolWave(int team)
+        {
+            Team = team;
+        }
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
 
+            float speedModifier = 0.02f + 0.01f * (PatternSpeed);
+            float directionModifier = -0.15f - 0.075f * (PatternComplexity);
```
Comment (Arrcival) [inline osu.Game.Rulesets.Vitaru/Objects/Projectiles/BulletPatterns.cs]: Change PatternSpeed and PatternComplexity to 
PatternComplexity - 1
(cause complexity of the patterns make more bullets, so speed depend from complexity)

File: osu.Game.Rulesets.Vitaru/Objects/Projectiles/BulletPatterns.cs [hunk 6]
```diff
@@ -7,61 +7,185 @@
...
+            for (int i = 1; i <= 3 + (PatternComplexity * 2); i++)
+            {
+                bulletAddRad(
+                    0.1f + Math.Abs(speedModifier),
```
Comment (Arrcival) [inline osu.Game.Rulesets.Vitaru/Objects/Projectiles/BulletPatterns.cs]: Change 0.1f to PatternSpeed
